# Data Wrangling Section
## EDA and Cleaning of Data
I have two datasets I'm using, one which contains three sections of information. The goal is to end up with 3 separate sets of data. I will upload each .csv file, separate and clean them, and then perform EDA to discover potential insights.

## Step 1: Upload, separate, save


In [1]:
# Import necessary modules
import pandas as pd
import chardet
import matplotlib.pyplot as plt

For some reason, only the PTI/Mortgage file was properly reading in. After further research, it seems that the Median Income file was corrupted, possibly with non UTF-8 characters. I researched online how to fix this issue, and the file properly read in. See code below.

In [2]:
# Read in working file
wide = pd.read_csv('Desktop/Capstone_Project_Data/Affordability_Wide_2018Q4_Public.csv')

In [3]:
# Read in corrupted file
# https://www.quora.com/How-do-I-fix-a-Unicode-error-while-reading-a-CSV-file-with-a-pandas-library-in-Python-3-6

with open(r'/Users/scoredy/Desktop/Capstone_Project_Data/Affordability_Income_2018Q4.csv', 'rb') as f:
    result = chardet.detect(f.read())
income = pd.read_csv(r'/Users/scoredy/Desktop/Capstone_Project_Data/Affordability_Income_2018Q4.csv', encoding=result['encoding'])
    

Inspect the wide dataframe, remove unnecessary info, and save as separate dataframes

In [4]:
wide.head()

RegionID                          RegionName  SizeRank            Index  \
0    102001                       United States         0  Price To Income   
1    394913                        New York, NY         1  Price To Income   
2    753899  Los Angeles-Long Beach-Anaheim, CA         2  Price To Income   
3    394463                         Chicago, IL         3  Price To Income   
4    394514               Dallas-Fort Worth, TX         4  Price To Income   

   HistoricAverage_1985thru1999   1979-03   1979-06   1979-09   1979-12  \
0                      2.786094  2.926690  2.962811  2.965507  2.970166   
1                      3.912069  2.942594  2.971022  2.981365  2.976273   
2                      4.479608  4.026750  4.148507  4.215555  4.285801   
3                      2.958478  3.035758  3.045651  2.984501  2.973171   
4                      3.068857  3.887457  4.025835  4.101038  4.130605   

    1980-03    ...      2016-09   2016-12   2017-03   2017-06   2017-09  \
0  2.957911    ...     3.297768  3.319031  3.334691  3.357272  3.371671   
1  3.021777    ...     5.298151  5.325995  5.327888  5.388576  5.401254   
2  4.316137    ...     8.631374  8.685383  8.671194  8.659377  8.714268   
3  2.895068    ...     3.026700  3.045975  3.055278  3.063627  3.072588   
4  4.135414    ...     2.937796  2.997831  3.036913  3.076033  3.110652   

    2017-12   2018-03   2018-06   2018-09   2018-12  
0  3.383362  3.435770  3.472983  3.501696  3.542629  
1  5.414706  5.468421  5.482805  5.477894  5.509601  
2  8.787902  8.898335  8.893782  8.879713  8.857709  
3  3.088122  3.131229  3.132547  3.123981  3.133726  
4  3.148341  3.261610  3.314444  3.346594  3.420982  

[5 rows x 165 columns]

In [5]:
wide.Index.describe()

count                1043
unique                  3
top       Price To Income
freq                  370
Name: Index, dtype: object

In [6]:
# From above, it seems there are 3 unique  values - let's see what values they are
wide.Index.value_counts()

Price To Income           370
Mortgage Affordability    370
Rent Affordability        303
Name: Index, dtype: int64

In [7]:
# 'Price To Income' and 'Mortgage Affordability' are the two descrpitors I need to separate
# Going to create two new df to encompass these - should each have 370 entries each
pti = wide[wide.Index == 'Price To Income'].reset_index()
mortgage = wide[wide.Index == 'Mortgage Affordability'].reset_index()

## Step 2: Inspect and clean all 3 dataframes: income, pti, and mortgage
Start with income df

In [8]:
income.describe()

RegionID    SizeRank       1979-03       1979-06       1979-09  \
count     934.000000  934.000000    382.000000    382.000000    382.000000   
mean   421158.204497  466.500000  15752.660759  16057.733586  16376.918927   
std     95976.022904  269.766875   2396.717700   2443.772105   2488.561923   
min    102001.000000    0.000000  10726.000000  10911.100000  11247.270000   
25%    394550.250000  233.250000  14092.477500  14380.897500  14718.507500   
50%    394809.500000  466.500000  15575.635000  15916.245000  16213.680000   
75%    395064.500000  699.750000  17336.695000  17711.447500  18077.687500   
max    786265.000000  933.000000  26197.160000  26786.470000  27393.020000   

            1979-12       1980-03       1980-06       1980-09       1980-12  \
count    382.000000    382.000000    382.000000    382.000000    382.000000   
mean   16686.666754  17065.960236  17356.238665  17584.245707  17790.123508   
std     2536.150289   2606.530962   2664.823462   2725.946506   2798.754494   
min    11503.600000  11754.170000  11993.670000  12226.950000  12409.510000   
25%    14991.237500  15267.665000  15579.115000  15759.737500  15866.087500   
50%    16502.270000  16842.490000  17095.080000  17265.800000  17451.870000   
75%    18376.837500  18758.410000  19091.230000  19291.957500  19546.305000   
max    27906.240000  28673.520000  29501.660000  30519.620000  31944.240000   

           ...              2016-09        2016-12        2017-03  \
count      ...           382.000000     382.000000     382.000000   
mean       ...         54505.689921   54875.245393   55430.101414   
std        ...         10285.648632   10426.495651   10611.134860   
min        ...         34682.010000   34895.570000   35194.460000   
25%        ...         47731.485000   47884.932500   48275.982500   
50%        ...         52590.165000   53035.245000   53376.150000   
75%        ...         59112.405000   59685.400000   60384.102500   
max        ...        110556.660000  112358.400000  114456.820000   

             2017-06        2017-09        2017-12        2018-03  \
count     382.000000     382.000000     382.000000     382.000000   
mean    55960.331047   56452.003586   56882.330157   57300.570014   
std     10814.075844   11017.856731   11227.804549   11330.239515   
min     35428.670000   35673.680000   35875.200000   36104.279257   
25%     48790.420000   49322.582500   49525.927500   49858.722789   
50%     53934.835000   54376.530000   54400.155000   54793.867096   
75%     61105.120000   61839.217500   62390.080000   62870.295932   
max    116570.370000  118454.770000  120414.690000  121388.698967   

             2018-06        2018-09        2018-12  
count     382.000000     382.000000     382.000000  
mean    57718.809871   58137.049729   58555.289586  
std     11432.835621   11535.588566   11638.494197  
min     36333.358514   36562.437771   36791.517028  
25%     50175.073078   50491.423366   50807.773655  
50%     55187.579192   55581.291288   55962.065107  
75%     63343.285176   63806.293436   64299.581248  
max    122362.707935  123336.716902  124310.725869  

[8 rows x 162 columns]

It looks like there are 934 entries, but most rows only have a count of 382. I suspect something is happening here, or there are many rows that have all or mostly NaN values. 

I'm going to look through rows that have all but 3 NaN values, so the "counted" amount will be only 3.

In [9]:
# Look through rows that have all but 3 NaN values
counted = income.apply(lambda x: x.count(), axis=1)

In [10]:
# Make Counted column with those values
income['Counted'] = counted
income.tail()

RegionID     RegionName  SizeRank  1979-03  1979-06  1979-09  1979-12  \
929    394767     Lamesa, TX       929      NaN      NaN      NaN      NaN   
930    753874      Craig, CO       930      NaN      NaN      NaN      NaN   
931    394968      Pecos, TX       931      NaN      NaN      NaN      NaN   
932    395188     Vernon, TX       932      NaN      NaN      NaN      NaN   
933    394743  Ketchikan, AK       933      NaN      NaN      NaN      NaN   

     1980-03  1980-06  1980-09   ...     2016-12  2017-03  2017-06  2017-09  \
929      NaN      NaN      NaN   ...         NaN      NaN      NaN      NaN   
930      NaN      NaN      NaN   ...         NaN      NaN      NaN      NaN   
931      NaN      NaN      NaN   ...         NaN      NaN      NaN      NaN   
932      NaN      NaN      NaN   ...         NaN      NaN      NaN      NaN   
933      NaN      NaN      NaN   ...         NaN      NaN      NaN      NaN   

     2017-12  2018-03  2018-06  2018-09  2018-12  Counted  
929      NaN      NaN      NaN      NaN      NaN        3  
930      NaN      NaN      NaN      NaN      NaN        3  
931      NaN      NaN      NaN      NaN      NaN        3  
932      NaN      NaN      NaN      NaN      NaN        3  
933      NaN      NaN      NaN      NaN      NaN        3  

[5 rows x 164 columns]

In [11]:
# Get rid of rows with all NaN values and save as new df
income_new = income[income.Counted > 3].reset_index()
income_new.tail()

index  RegionID       RegionName  SizeRank   1979-03   1979-06   1979-09  \
377    456    394489     Columbus, IN       456  18949.12  19301.27  19685.85   
378    459    394445       Casper, WY       459  21962.63  22433.77  23114.04   
379    512    395199  Walla Walla, WA       512  15092.45  15513.87  15762.96   
380    525    394787     Lewiston, ID       525  15389.83  15728.34  15828.83   
381    552    394444  Carson City, NV       552  18248.86  18481.70  18794.85   

      1979-12   1980-03   1980-06   ...      2016-12   2017-03   2017-06  \
377  20079.76  20543.30  20846.30   ...     59804.53  60603.54  61283.24   
378  23957.56  24835.12  25601.92   ...     59004.07  59452.40  60165.67   
379  15738.86  15757.61  15722.60   ...     52633.97  53278.18  53923.35   
380  15723.02  15740.12  15625.26   ...     53198.41  54112.23  55120.84   
381  19178.59  19723.67  20143.51   ...     51622.53  52229.23  52534.59   

      2017-09   2017-12       2018-03       2018-06       2018-09  \
377  61854.25  62390.97  62852.226198  63313.482397  63774.738595   
378  60927.64  61638.29  62136.869095  62635.448190  63134.027286   
379  54425.38  55000.97  55445.861217  55890.752434  56335.643651   
380  55603.19  55615.27  56065.130160  56514.990320  56964.850480   
381  53168.78  53510.82  53943.657709  54376.495418  54809.333126   

          2018-12  Counted  
377  64235.994794      163  
378  63632.606381      163  
379  56780.534868      163  
380  57414.710640      163  
381  55242.170835      163  

[5 rows x 165 columns]

In [12]:
# Drop the 'index' and 'Counted' columns
income_new = income_new.drop(['index', 'Counted'], axis=1)

I personally hate how these years are stacked. I want to transform the data so we have a column for year and a column for quarter, then make a fourth table with the summary detail that's similar among them all.

In [14]:
income_info = income_new.iloc[:,0:3]

In [15]:
income_info.head()

RegionID                          RegionName  SizeRank
0    102001                       United States         0
1    394913                        New York, NY         1
2    753899  Los Angeles-Long Beach-Anaheim, CA         2
3    394463                         Chicago, IL         3
4    394514               Dallas-Fort Worth, TX         4

In [16]:
income_data = income_new.drop(['RegionID', 'SizeRank'], axis=1)

In [18]:
income_data = income_data.set_index('RegionName').T.rename_axis('YearQtr').rename_axis(None, 1).reset_index()

In [19]:
income_data.head()

YearQtr  United States  New York, NY  Los Angeles-Long Beach-Anaheim, CA  \
0  1979-03       16346.86      17536.34                            17913.83   
1  1979-06       16675.92      17888.75                            18285.02   
2  1979-09       17015.62      18255.17                            18746.27   
3  1979-12       17325.61      18614.93                            19213.79   
4  1980-03       17700.06      19075.72                            19782.36   

   Chicago, IL  Dallas-Fort Worth, TX  Philadelphia, PA  Houston, TX  \
0     20056.35               18147.33          17530.66     20209.30   
1     20460.12               18458.77          17869.90     20569.30   
2     20910.06               18899.16          18248.18     21125.56   
3     21267.37               19454.80          18589.76     21866.45   
4     21678.20               20069.21          19067.80     22677.02   

   Washington, DC  Miami-Fort Lauderdale, FL       ...         \
0        22611.34                   15673.84       ...          
1        23095.61                   15963.28       ...          
2        23441.83                   16269.95       ...          
3        23673.91                   16509.61       ...          
4        24067.25                   16908.27       ...          

   Grants Pass, OR  Grand Island, NE  Danville, IL  Great Falls, MT  \
0         12689.42          15623.14      15719.82         15854.94   
1         13003.88          16037.15      16056.81         16024.79   
2         13234.73          16226.53      16391.74         16015.16   
3         13371.97          16610.36      16603.64         16305.12   
4         13520.85          16841.63      16822.21         16972.48   

   Hinesville, GA  Columbus, IN  Casper, WY  Walla Walla, WA  Lewiston, ID  \
0        11142.75      18949.12    21962.63         15092.45      15389.83   
1        11333.70      19301.27    22433.77         15513.87      15728.34   
2        11663.90      19685.85    23114.04         15762.96      15828.83   
3        12113.35      20079.76    23957.56         15738.86      15723.02   
4        12588.44      20543.30    24835.12         15757.61      15740.12   

   Carson City, NV  
0         18248.86  
1         18481.70  
2         18794.85  
3         19178.59  
4         19723.67  

[5 rows x 383 columns]

In [20]:
# Split up column with Year and Quarter within it
new = income_data['YearQtr'].str.split('-', n=1, expand=True)

In [21]:
# Add Year and Qtr columns (separated from before)
income_data['Year'] = new[0]
income_data['Qtr'] = new[1]

In [22]:
income_data.head()

YearQtr  United States  New York, NY  Los Angeles-Long Beach-Anaheim, CA  \
0  1979-03       16346.86      17536.34                            17913.83   
1  1979-06       16675.92      17888.75                            18285.02   
2  1979-09       17015.62      18255.17                            18746.27   
3  1979-12       17325.61      18614.93                            19213.79   
4  1980-03       17700.06      19075.72                            19782.36   

   Chicago, IL  Dallas-Fort Worth, TX  Philadelphia, PA  Houston, TX  \
0     20056.35               18147.33          17530.66     20209.30   
1     20460.12               18458.77          17869.90     20569.30   
2     20910.06               18899.16          18248.18     21125.56   
3     21267.37               19454.80          18589.76     21866.45   
4     21678.20               20069.21          19067.80     22677.02   

   Washington, DC  Miami-Fort Lauderdale, FL ...   Danville, IL  \
0        22611.34                   15673.84 ...       15719.82   
1        23095.61                   15963.28 ...       16056.81   
2        23441.83                   16269.95 ...       16391.74   
3        23673.91                   16509.61 ...       16603.64   
4        24067.25                   16908.27 ...       16822.21   

   Great Falls, MT  Hinesville, GA  Columbus, IN  Casper, WY  Walla Walla, WA  \
0         15854.94        11142.75      18949.12    21962.63         15092.45   
1         16024.79        11333.70      19301.27    22433.77         15513.87   
2         16015.16        11663.90      19685.85    23114.04         15762.96   
3         16305.12        12113.35      20079.76    23957.56         15738.86   
4         16972.48        12588.44      20543.30    24835.12         15757.61   

   Lewiston, ID  Carson City, NV  Year  Qtr  
0      15389.83         18248.86  1979   03  
1      15728.34         18481.70  1979   06  
2      15828.83         18794.85  1979   09  
3      15723.02         19178.59  1979   12  
4      15740.12         19723.67  1980   03  

[5 rows x 385 columns]

In [23]:
# Make my own dataframe to merge onto for the quarters
num = ['03', '06', '09', '12']
qtr = ['Q1', 'Q2', 'Q3', 'Q4']

In [24]:
qtr_list = pd.DataFrame({'Number':num, 'Quarter':qtr})

In [26]:
# Merge the two df to fill in Qtr correctly
income_data = pd.merge(income_data, qtr_list, how='left', left_on='Qtr', right_on='Number')

In [27]:
income_data.head(50)

YearQtr  United States  New York, NY  Los Angeles-Long Beach-Anaheim, CA  \
0   1979-03       16346.86      17536.34                            17913.83   
1   1979-06       16675.92      17888.75                            18285.02   
2   1979-09       17015.62      18255.17                            18746.27   
3   1979-12       17325.61      18614.93                            19213.79   
4   1980-03       17700.06      19075.72                            19782.36   
5   1980-06       18003.37      19493.98                            20306.08   
6   1980-09       18309.93      19894.97                            20737.44   
7   1980-12       18655.45      20320.32                            21091.91   
8   1981-03       19069.98      20744.95                            21468.44   
9   1981-06       19406.03      21122.72                            21765.04   
10  1981-09       19799.97      21558.15                            22166.95   
11  1981-12       20170.18      22022.97                            22582.74   
12  1982-03       20502.31      22477.92                            22992.00   
13  1982-06       20700.73      22789.32                            23190.23   
14  1982-09       20881.75      23220.66                            23452.38   
15  1982-12       21040.41      23734.20                            23705.82   
16  1983-03       21193.80      24248.75                            24117.49   
17  1983-06       21375.66      24737.60                            24585.77   
18  1983-09       21630.27      25165.92                            25030.31   
19  1983-12       22021.07      25623.99                            25389.37   
20  1984-03       22488.88      26116.08                            25864.12   
21  1984-06       22990.99      26666.66                            26337.42   
22  1984-09       23432.67      27269.79                            26881.19   
23  1984-12       23771.43      27970.76                            27419.67   
24  1985-03       24066.18      28541.66                            27979.61   
25  1985-06       24283.44      28989.64                            28356.13   
26  1985-09       24553.00      29415.56                            28777.79   
27  1985-12       24893.82      29923.29                            29250.52   
28  1986-03       25286.77      30544.08                            29844.92   
29  1986-06       25669.88      31165.14                            30422.71   
30  1986-09       26001.75      31825.38                            30978.42   
31  1986-12       26265.34      32424.83                            31455.41   
32  1987-03       26569.60      33117.27                            32025.05   
33  1987-06       26910.13      33776.37                            32634.48   
34  1987-09       27213.02      34316.56                            33185.84   
35  1987-12       27481.43      34758.98                            33694.81   
36  1988-03       27769.24      35542.84                            34236.26   
37  1988-06       28053.96      36136.27                            34857.09   
38  1988-09       28411.98      36674.40                            35505.66   
39  1988-12       28889.20      37123.56                            36124.99   
40  1989-03       29475.01      37818.46                            36850.97   
41  1989-06       29977.60      38547.66                            37378.23   
42  1989-09       30315.07      38893.13                            37564.65   
43  1989-12       30456.32      38823.75                            37507.90   
44  1990-03       30703.91      38933.73                            37703.90   
45  1990-06       31018.24      39009.43                            37975.53   
46  1990-09       31294.84      39504.59                            38175.79   
47  1990-12       31392.28      39990.00                            38390.29   
48  1991-03       31325.60      40080.54                            38421.99   
49  

In [28]:
#Remove redundant columns
income_data = income_data.drop(['YearQtr', 'Qtr', 'Number'], axis=1)

In [29]:
income_data.head()

United States  New York, NY  Los Angeles-Long Beach-Anaheim, CA  \
0       16346.86      17536.34                            17913.83   
1       16675.92      17888.75                            18285.02   
2       17015.62      18255.17                            18746.27   
3       17325.61      18614.93                            19213.79   
4       17700.06      19075.72                            19782.36   

   Chicago, IL  Dallas-Fort Worth, TX  Philadelphia, PA  Houston, TX  \
0     20056.35               18147.33          17530.66     20209.30   
1     20460.12               18458.77          17869.90     20569.30   
2     20910.06               18899.16          18248.18     21125.56   
3     21267.37               19454.80          18589.76     21866.45   
4     21678.20               20069.21          19067.80     22677.02   

   Washington, DC  Miami-Fort Lauderdale, FL  Atlanta, GA   ...     \
0        22611.34                   15673.84     17097.70   ...      
1        23095.61                   15963.28     17419.05   ...      
2        23441.83                   16269.95     17787.09   ...      
3        23673.91                   16509.61     18160.24   ...      
4        24067.25                   16908.27     18627.02   ...      

   Danville, IL  Great Falls, MT  Hinesville, GA  Columbus, IN  Casper, WY  \
0      15719.82         15854.94        11142.75      18949.12    21962.63   
1      16056.81         16024.79        11333.70      19301.27    22433.77   
2      16391.74         16015.16        11663.90      19685.85    23114.04   
3      16603.64         16305.12        12113.35      20079.76    23957.56   
4      16822.21         16972.48        12588.44      20543.30    24835.12   

   Walla Walla, WA  Lewiston, ID  Carson City, NV  Year  Quarter  
0         15092.45      15389.83         18248.86  1979       Q1  
1         15513.87      15728.34         18481.70  1979       Q2  
2         15762.96      15828.83         18794.85  1979       Q3  
3         15738.86      15723.02         19178.59  1979       Q4  
4         15757.61      15740.12         19723.67  1980       Q1  

[5 rows x 384 columns]

In [30]:
#Rearrange columns: first find column list, then rearrange, then recreate df
cols = income_data.columns.tolist()
cols

['United States',
 'New York, NY',
 'Los Angeles-Long Beach-Anaheim, CA',
 'Chicago, IL',
 'Dallas-Fort Worth, TX',
 'Philadelphia, PA',
 'Houston, TX',
 'Washington, DC',
 'Miami-Fort Lauderdale, FL',
 'Atlanta, GA',
 'Boston, MA',
 'San Francisco, CA',
 'Detroit, MI',
 'Riverside, CA',
 'Phoenix, AZ',
 'Seattle, WA',
 'Minneapolis-St Paul, MN',
 'San Diego, CA',
 'St. Louis, MO',
 'Tampa, FL',
 'Baltimore, MD',
 'Denver, CO',
 'Pittsburgh, PA',
 'Portland, OR',
 'Charlotte, NC',
 'Sacramento, CA',
 'San Antonio, TX',
 'Orlando, FL',
 'Cincinnati, OH',
 'Cleveland, OH',
 'Kansas City, MO',
 'Las Vegas, NV',
 'Columbus, OH',
 'Indianapolis, IN',
 'San Jose, CA',
 'Austin, TX',
 'Virginia Beach, VA',
 'Nashville, TN',
 'Providence, RI',
 'Milwaukee, WI',
 'Jacksonville, FL',
 'Memphis, TN',
 'Oklahoma City, OK',
 'Louisville-Jefferson County, KY',
 'Hartford, CT',
 'Richmond, VA',
 'New Orleans, LA',
 'Buffalo, NY',
 'Raleigh, NC',
 'Birmingham, AL',
 'Salt Lake City, UT',
 'Rochester, 

In [31]:
cols = cols[-2:] + cols[:-2]
cols

['Year',
 'Quarter',
 'United States',
 'New York, NY',
 'Los Angeles-Long Beach-Anaheim, CA',
 'Chicago, IL',
 'Dallas-Fort Worth, TX',
 'Philadelphia, PA',
 'Houston, TX',
 'Washington, DC',
 'Miami-Fort Lauderdale, FL',
 'Atlanta, GA',
 'Boston, MA',
 'San Francisco, CA',
 'Detroit, MI',
 'Riverside, CA',
 'Phoenix, AZ',
 'Seattle, WA',
 'Minneapolis-St Paul, MN',
 'San Diego, CA',
 'St. Louis, MO',
 'Tampa, FL',
 'Baltimore, MD',
 'Denver, CO',
 'Pittsburgh, PA',
 'Portland, OR',
 'Charlotte, NC',
 'Sacramento, CA',
 'San Antonio, TX',
 'Orlando, FL',
 'Cincinnati, OH',
 'Cleveland, OH',
 'Kansas City, MO',
 'Las Vegas, NV',
 'Columbus, OH',
 'Indianapolis, IN',
 'San Jose, CA',
 'Austin, TX',
 'Virginia Beach, VA',
 'Nashville, TN',
 'Providence, RI',
 'Milwaukee, WI',
 'Jacksonville, FL',
 'Memphis, TN',
 'Oklahoma City, OK',
 'Louisville-Jefferson County, KY',
 'Hartford, CT',
 'Richmond, VA',
 'New Orleans, LA',
 'Buffalo, NY',
 'Raleigh, NC',
 'Birmingham, AL',
 'Salt Lake Cit

In [32]:
# Organized df with Year and Qtr columns first
income_data = income_data[cols]
income_data

Year Quarter  United States  New York, NY  \
0    1979      Q1   16346.860000  17536.340000   
1    1979      Q2   16675.920000  17888.750000   
2    1979      Q3   17015.620000  18255.170000   
3    1979      Q4   17325.610000  18614.930000   
4    1980      Q1   17700.060000  19075.720000   
5    1980      Q2   18003.370000  19493.980000   
6    1980      Q3   18309.930000  19894.970000   
7    1980      Q4   18655.450000  20320.320000   
8    1981      Q1   19069.980000  20744.950000   
9    1981      Q2   19406.030000  21122.720000   
10   1981      Q3   19799.970000  21558.150000   
11   1981      Q4   20170.180000  22022.970000   
12   1982      Q1   20502.310000  22477.920000   
13   1982      Q2   20700.730000  22789.320000   
14   1982      Q3   20881.750000  23220.660000   
15   1982      Q4   21040.410000  23734.200000   
16   1983      Q1   21193.800000  24248.750000   
17   1983      Q2   21375.660000  24737.600000   
18   1983      Q3   21630.270000  25165.920000   
19   1983      Q4   22021.070000  25623.990000   
20   1984      Q1   22488.880000  26116.080000   
21   1984      Q2   22990.990000  26666.660000   
22   1984      Q3   23432.670000  27269.790000   
23   1984      Q4   23771.430000  27970.760000   
24   1985      Q1   24066.180000  28541.660000   
25   1985      Q2   24283.440000  28989.640000   
26   1985      Q3   24553.000000  29415.560000   
27   1985      Q4   24893.820000  29923.290000   
28   1986      Q1   25286.770000  30544.080000   
29   1986      Q2   25669.880000  31165.140000   
..    ...     ...            ...           ...   
130  2011      Q3   50594.090000  64635.040000   
131  2011      Q4   50570.770000  64249.800000   
132  2012      Q1   50815.670000  64396.230000   
133  2012      Q2   51099.870000  64751.960000   
134  2012      Q3   51421.410000  65217.350000   
135  2012      Q4   52147.060000  66125.470000   
136  2013      Q1   52179.770000  66087.020000   
137  2013      Q2   52411.950000  66396.690000   
138  2013      Q3   52095.690000  65954.700000   
139  2013      Q4   52312.600000  66198.360000   
140  2014      Q1   52743.340000  66509.810000   
141  2014      Q2   53328.940000  66937.990000   
142  2014      Q3   54015.650000  67471.230000   
143  2014      Q4   54540.070000  67874.610000   
144  2015      Q1   55010.770000  68243.810000   
145  2015      Q2   55456.900000  68750.330000   
146  2015      Q3   56026.510000  69559.020000   
147  2015      Q4   56605.810000  70388.880000   
148  2016      Q1   57017.050000  71139.930000   
149  2016      Q2   57324.210000  71694.190000   
150  2016      Q3   57766.340000  72364.870000   
151  2016      Q4   58360.400000  72812.680000   
152  2017      Q1   59165.910000  73819.120000   
153  2017      Q2   59959.390000  74936.310000   
154  2017      Q3   60741.390000  75963.840000   
155  2017      Q4   61477.310000  76753.930000   
156  2018      Q1   61936.623236  77408.082812   
157  2018      Q2   62395.936471  78062.235625   
158  2018      Q3   62855.249707  78716.388438   
159  2018      Q4   63314.562943  79370.541250   

     Los Angeles-Long Beach-Anaheim, CA   Chicago, IL  Dallas-Fort Worth, TX  \
0                          17913.830000  20056.350000           18147.330000   
1                          18285.020000  20460.120000           18458.770000   
2                          18746.270000  20910.060000           18899.160000   
3                          19213.790000  21267.370000           19454.800000   
4                          19782.360000  21678.200000           20069.210000   
5                          20306.080000  21848.900000           20608.070000   
6                          20737.440000  21984.030000           21021.310000   
7                          21091.910000  22067.130000           21363.170000   
8                          21468.440000  22395.990000           21890.500000   
9                          21765.040000  22598.730000           22279.250000   
10                      

Now we do the tranposing and separating to the pti df

In [33]:
pti.describe()

index       RegionID    SizeRank  HistoricAverage_1985thru1999  \
count  370.00000     370.000000  370.000000                    208.000000   
mean   184.50000  401742.972973  195.070270                      2.711890   
std    106.95404   54586.758478  122.513427                      0.747161   
min      0.00000  102001.000000    0.000000                      1.394539   
25%     92.25000  394531.250000   92.250000                      2.254492   
50%    184.50000  394771.000000  184.500000                      2.591051   
75%    276.75000  395021.250000  289.750000                      3.048728   
max    369.00000  753924.000000  552.000000                      5.367640   

          1979-03     1979-06     1979-09     1979-12     1980-03     1980-06  \
count  102.000000  112.000000  122.000000  127.000000  126.000000  128.000000   
mean     3.062694    3.107583    3.110520    3.102669    3.090519    3.059573   
std      0.681218    0.720530    0.726451    0.746350    0.758582    0.757340   
min      1.494261    1.535356    1.611391    1.626672    1.547194    1.537382   
25%      2.664154    2.673084    2.602155    2.598742    2.559139    2.586727   
50%      2.938343    2.996845    2.985558    2.973125    2.960303    2.946818   
75%      3.439354    3.501884    3.522286    3.534572    3.524095    3.433256   
max      4.926503    5.103076    5.078746    5.173311    5.361290    5.278691   

          ...         2016-09     2016-12     2017-03     2017-06     2017-09  \
count     ...      370.000000  370.000000  370.000000  370.000000  370.000000   
mean      ...        3.202486    3.231632    3.252624    3.273904    3.289188   
std       ...        1.282708    1.297437    1.304790    1.310177    1.315597   
min       ...        1.283619    1.293059    1.290162    1.268604    1.261793   
25%       ...        2.373552    2.402997    2.406304    2.428860    2.435601   
50%       ...        2.890529    2.901130    2.892676    2.909188    2.911176   
75%       ...        3.619118    3.667975    3.712947    3.755117    3.803021   
max       ...        9.759626    9.802021    9.820207    9.813742    9.909852   

          2017-12     2018-03     2018-06     2018-09     2018-12  
count  370.000000  370.000000  370.000000  370.000000  370.000000  
mean     3.306093    3.351524    3.386901    3.417588    3.456356  
std      1.326680    1.354087    1.369169    1.374926    1.371738  
min      1.288702    1.294976    1.304308    1.294875    1.334133  
25%      2.450004    2.483608    2.505901    2.517198    2.545691  
50%      2.895304    2.934259    2.969003    3.012085    3.045951  
75%      3.833632    3.856949    3.863026    3.930462    4.000683  
max      9.973271   10.086060   10.361000   10.340797   10.077972  

[8 rows x 164 columns]